src/core/response_cache.cc

```diff
@@ -195,53 +167,135 @@ RequestResponseCache::Lookup(
   // Populate passed-in "ptr" from cache entry
   auto entry = iter->second;
   // Build InferenceResponse from CacheEntry
-  RETURN_IF_ERROR(BuildInferenceResponse(entry, ptr));
```

Comment:
Is this breakdown necessary?
I believe it only adds the following message before error:
"Failed to build inference response: "